x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/signals/query_signals_route.test.ts

```diff
@@ -4,69 +4,55 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ServerInjectOptions } from 'hapi';
-
-import { querySignalsRoute } from './query_signals_route';
-import * as myUtils from '../utils';
+import { DETECTION_ENGINE_QUERY_SIGNALS_URL } from '../../../../../common/constants';
 import {
   getSignalsQueryRequest,
   getSignalsAggsQueryRequest,
   typicalSignalsQuery,
   typicalSignalsQueryAggs,
+  getSignalsAggsAndQueryRequest,
 } from '../__mocks__/request_responses';
-import { createMockServer, createMockConfig, clientsServiceMock } from '../__mocks__';
-import { DETECTION_ENGINE_QUERY_SIGNALS_URL } from '../../../../../common/constants';
+import { requestContextMock, serverMock, requestMock } from '../__mocks__';
+import { querySignalsRoute } from './query_signals_route';
 
 describe('query for signal', () => {
-  let server = createMockServer();
-  let config = createMockConfig();
-  let getClients = clientsServiceMock.createGetScoped();
-  let clients = clientsServiceMock.createClients();
+  let server: ReturnType<typeof serverMock.create>;
+  let { clients, context } = requestContextMock.createTools();
 
   beforeEach(() => {
-    jest.resetAllMocks();
-    jest.spyOn(myUtils, 'getIndex').mockReturnValue('fakeindex');
+    server = serverMock.create();
+    ({ clients, context } = requestContextMock.createTools());
 
-    server = createMockServer();
-    config = createMockConfig();
-    getClients = clientsServiceMock.createGetScoped();
-    clients = clientsServiceMock.createClients();
+    // clients.clusterClient.callAsCurrentUser.mockResolvedValue(true); // TODO: what is the return value of a successful query?
```

Comment:
You still need this commented out TODO here?

Reply:
For this particular route, we push the result of this query directly into a response, so for better or worse these tests exercise all the appropriate behavior. My intention for all of these had been to mock everything required for the 'happy path', so this one technically does not do that.

I'll figure out what the return value needs to be here, and mock it appropriately.

Reply:
7087da2905c95edd80aa73c4f1a90005b7b8fe83